### PR TITLE
feat(tests): add full e2e coverage for Popover component

### PR DESCRIPTION
**Pull request by dmitry-zhemchugov**
# PR Checklist
Before creating new PR, please take a look at checklist below to make sure that you've done everything that needs to be done before we can merge it.

 - [ ] read and followed the [CONTRIBUTING.md](https://github.com/valor-software/ngx-bootstrap/blob/development/CONTRIBUTING.md) guide.
 - [ ] built and tested the changes locally.
 - [ ] added/updated tests.
 - [ ] added/updated API documentation.
 - [ ] added/updated demos.

### CODE REVIEW

**Comment by ludmilanesvitiy** [comment]
@dmitry-zhemchugov All PR labels, reviewers, assignees should be similar to https://github.com/valor-software/ngx-bootstrap/pull/4871

File: cypress/full/popover_page_spec.ts [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+
+describe('Popover demo page test suite', () => {
+
+  const popover = new PopoverPo();
+
+  before(() => popover.navigateTo());
+
+  describe('Placement popover', () => {
+
+    const basic = popover.exampleDemosArr.placement;
```
Comment (ludmilanesvitiy) [inline cypress/full/popover_page_spec.ts]: placementDemo

File: cypress/support/base.component.ts [hunk 2]
```diff
@@ -55,6 +55,18 @@ export abstract class BaseComponent {
     cy.get(`${ baseSelector } input`).eq(inputIndex ? inputIndex : 0).dblclick();
   }
 
+  typeOnInput(baseSelector: string, forInput: string, inputIndex?: number) {
```
Comment (ludmilanesvitiy) [inline cypress/support/base.component.ts]: same as clearInputAndSendKeys method, in this class

Comment (ludmilanesvitiy) [reply cypress/support/base.component.ts]: need to remove it

File: cypress/support/popover.po.ts [hunk 3]
```diff
@@ -5,7 +5,77 @@ export class PopoverPo extends BaseComponent {
...
-    basic: 'demo-popover-basic'
+    basic: 'demo-popover-basic',
+    placement: 'demo-popover-placement',
+    dismiss: 'demo-popover-dismiss',
+    dynamic: 'demo-popover-dynamic',
+    customContent: 'demo-popover-custom-content',
+    dynamicHtml: 'demo-popover-dynamic-html',
+    appendToBody: 'demo-popover-container',
+    visibilityEvents: 'demo-popover-events',
+    eventTextBlock: 'pre.card.card-block.card-header.mb-3',
```
Comment (ludmilanesvitiy) [inline cypress/support/popover.po.ts]: 1. bad selector, need to find more stable
2. it should be not in the exampleDemosArr

File: cypress/support/popover.po.ts [hunk 4]
```diff
@@ -5,7 +5,77 @@ export class PopoverPo extends BaseComponent {
...
+  isPopoverDismissAfterClick(baseSelector: string) {
+    cy.get(`${baseSelector}`).closest('.section.bd-example').click('topRight');
+    cy.get(`${baseSelector}`).should('not.to.have.descendants', this.containerPopover);
+  }
+
+  isPopoverDismiss(baseSelector: string) {
+    cy.get(`${baseSelector}`).should('not.to.have.descendants', this.containerPopover);
+  }
+
+  isEventShown(baseSelector1: string) {
```
Comment (ludmilanesvitiy) [inline cypress/support/popover.po.ts]: what does it mean baseSelector1 ?

File: cypress/support/popover.po.ts [hunk 5]
```diff
@@ -5,7 +5,77 @@ export class PopoverPo extends BaseComponent {
...
+    cy.get(`${baseSelector1}`).contains('shown');
+  }
+
+  isEventHidden(baseSelector1: string) {
```
Comment (ludmilanesvitiy) [inline cypress/support/popover.po.ts]: what does it mean baseSelector1 ?

**Review by ludmilanesvitiy — CHANGES_REQUESTED** [review]

File: cypress/full/popover_page_spec.ts [hunk 6]
```diff
@@ -0,0 +1,348 @@
...
+    const configuringDefaults = popover.exampleDemosArr.configuringDefaults;
+
+    it('when user clicks on "Preconfigured popover", then popover-container shown', () => {
+        popover.clickOnBtn(configuringDefaults);
+        popover.isPopoverAppears(popover.body);
+        popover.isPopoverVisible(popover.body);
+    });
+
+    it('when user clicks on "Preconfigured popover" again, nothing happens', () => {
+        popover.clickOnBtn(configuringDefaults);
```
Comment (ludmilanesvitiy) [inline cypress/full/popover_page_spec.ts]: 2

File: cypress/integration/popover_page_spec.ts [hunk 7]
```diff
@@ -1,18 +1,24 @@
 import { PopoverPo } from '../support/popover.po';
 
 describe('Popover demo page test suite', () => {
+
   const popover = new PopoverPo();
 
-  beforeEach(() => popover.navigateTo());
+  before(() => popover.navigateTo());
```
Comment (ludmilanesvitiy) [inline cypress/integration/popover_page_spec.ts]: beforeEach

File: cypress/integration/popover_page_spec.ts [hunk 8]
```diff
@@ -1,18 +1,24 @@
...
+    it('when user clicks on "Live demo", then popover-container shown', () => {
+      popover.clickOnBtn(basicPopover);
+      popover.isPopoverAppears(basicPopover);
+      popover.isPopoverVisible(basicPopover);
+    });
 
-      popover.clickByText(basicPopover, buttonText);
-      cy.get(basicPopover).should('to.have.descendants', 'popover-container');
+    it('when user clicks on "Live demo" again, then popover-container disappeared', () => {
+      popover.clickOnBtn(basicPopover);
```
Comment (ludmilanesvitiy) [inline cypress/integration/popover_page_spec.ts:20]: 2

**Review by valorkin — APPROVED** [review]

**Pull request merged**